### PR TITLE
Move the region_scope_tree query to librustc_passes.

### PR DESCRIPTION
**Pull request by cjgillot**
Split out of #67688.

r? @Zoxc

### CODE REVIEW

File: src/librustc/middle/region.rs [hunk 1]
```diff
@@ -8,22 +8,17 @@
```
Comment (Zoxc) [inline src/librustc/middle/region.rs]: The `This file builds up the ScopeTree` comment is now outdated.

Comment (Zoxc) [reply src/librustc/middle/region.rs]: Maybe just change it to `This file declares ScopeTree`? The rest is still relevant.

**Comment by cjgillot** [comment]
Rebased.

**Pull request merged**